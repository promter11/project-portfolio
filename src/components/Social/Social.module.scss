@import "../../styles/_variables";

.social {
  display: flex;

  .icon {
    path {
      transition: fill 0.2s ease-in-out;
    }

    &:hover {
      path {
        fill: $social-fill-color;
      }
    }
  }

  .link {
    display: block;
    width: 2rem;
    height: 2rem;

    &:not(:last-child) {
      margin-right: 3rem;
    }
  }
}
