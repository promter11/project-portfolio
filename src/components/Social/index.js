import React, { createElement } from "react";

import styles from "./Social.module.scss";

const Social = ({ socials }) => {
  return (
    <div className={styles.social}>
      {socials.map(({ path, component }, index) => {
        return (
          <a key={index} className={styles.link} href={path}>
            {createElement(component, { className: styles.icon }, null)}
          </a>
        );
      })}
    </div>
  );
};

export default Social;
