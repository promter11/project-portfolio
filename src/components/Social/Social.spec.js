import React from "react";
import { shallow } from "enzyme";

import Social from "./index";

import { ReactComponent as MockComponent } from "../../assets/icons/aside/telegram.svg";

const props = {
  socials: [
    {
      path: "https://link.com",
      component: MockComponent,
    },
    {
      path: "https://path.ru",
      component: MockComponent,
    },
    {
      path: "mailto:example@gmail.com",
      component: MockComponent,
    },
  ],
};

describe("Social component", () => {
  it("renders correctly with items", () => {
    const wrapper = shallow(<Social {...props} />);

    expect(wrapper).toMatchSnapshot();
  });
});
