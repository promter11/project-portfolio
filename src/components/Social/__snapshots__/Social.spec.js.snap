// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Social component renders correctly with items 1`] = `
<div
  className="social"
>
  <a
    className="link"
    href="https://link.com"
    key="0"
  >
    <ForwardRef(SvgTelegram)
      className="icon"
    />
  </a>
  <a
    className="link"
    href="https://path.ru"
    key="1"
  >
    <ForwardRef(SvgTelegram)
      className="icon"
    />
  </a>
  <a
    className="link"
    href="mailto:example@gmail.com"
    key="2"
  >
    <ForwardRef(SvgTelegram)
      className="icon"
    />
  </a>
</div>
`;
